nestkernel/event_delivery_manager.cpp

```diff
@@ -826,18 +814,21 @@ EventDeliveryManager::gather_target_data( const size_t tid )
       set_complete_marker_target_data_( assigned_ranks, send_buffer_position );
     }
     kernel().connection_manager.save_source_table_entry_point( tid );
+    kernel().get_omp_synchronization_stopwatch().start();
 #pragma omp barrier
+    kernel().get_omp_synchronization_stopwatch().stop();
     kernel().connection_manager.clean_source_table( tid );
 
 #pragma omp master
     {
-#ifdef TIMER_DETAILED
-      sw_communicate_target_data_.start();
+#if defined( HAVE_MPI ) && defined( TIMER_DETAILED )
+      kernel().get_mpi_synchronization_stopwatch().start();
```

Comment:
Since we are only really interested in the synchronization during spike communication we could remove these lines. Documentation of the timer might have to be adjusted to state that it only measures synchronization during gather spike data.